Apps/Sandcastle/gallery/development/Picking Sublayer Regions.html

```diff
@@ -0,0 +1,174 @@
+                stack.exterior.width,
+                stack.exterior.height),
+        material : Cesium.Color.DARKGREEN,
+        fill: true,
+        outline : false,
+        outlineColor : Cesium.Color.BLACK
+    },
+    label: {
+        text: 'Cesium Tower',
+        show: false
+    }
+});
+
+//collection for hittable entity testing
+var hittable = new Cesium.EntityCollection();
+
+//Non-visible "pick-able" Objects
+var interior = new Cesium.Entity({
+    parent: stack.entity,
+    id: 'stack-blocks',
+    show: true
+});
+
+for (var box = 0; box < stack.boxes; box++) {
+    var heightInStack = (box * stack.interior.height) + (box * stack.interior.spacing);
+    entities.add(
+            hittable.add({
+                parent : interior,
+                id: 'stack-blocks-' + box,
+                name: 'Contained Box ' + (box+1),
+                label: {
+                    text: 'Contained Box: ' + (box+1),
+                    show: false,
+                    eyeOffset: new Cesium.Cartesian3(25.0, 0.0, -25.0)
+                },
+                position : Cesium.Cartesian3.fromDegrees(
+                        stack.location.lon,
+                        stack.location.lat,
+                        heightInStack + offsetInteriorHeight),
+                box : {
+                    dimensions : new Cesium.Cartesian3(
+                            stack.interior.length,
+                            stack.interior.width,
+                            stack.interior.height),
+                    material : Cesium.Color.WHITE.withAlpha(0.004),
+                    outline : false,
+                    outlineColor : Cesium.Color.BLUE
+                }
+            })
+    );
+}
+
+entities.add(interior);
+
+viewer.trackedEntity = stack.entity;
+viewer.zoomTo(viewer.entities);
+
+//UI Interactions
+Sandcastle.addToolbarButton('Toggle Stack Exterior Fill', function (){
+    exterior.box.fill = !exterior.box.fill.getValue();
+    exterior.box.outline = !exterior.box.outline.getValue();
+});
+
+Sandcastle.addToolbarButton('Toggle Stack Interior Outlines', function (){
+    var boxes = hittable.values;
+    for (var i = 0; i < boxes.length; i++) {
+        boxes[i].box.outline = !boxes[i].box.outline.getValue();
+    }
+});
+
+handler.setInputAction( function searchHittable(movement) {
+    //flip the shown flags
+    interior.show = true;
+    var currentObjects = scene.drillPick(movement.position);
+    console.log(currentObjects.length + ' entites found');
+
+    //set them back to original state
+    for (var i = 0; i < currentObjects.length; i++) {
+        var current = currentObjects[i].id;
+        if (pickedObject === current) {
+            return; //picked object hasn't changed so do nothing
+        }
+
+        if (hittable.contains(current)) {
+            console.log('picked: ' + current.name);
+
+            //clear the prior selected object
+            ((pickedObject || {}).label || {}).show = false;
```

Comment:
Not sure how much we care about consistency in demos, but our usual style is more along the lines of:

```
if (Cesium.defined(pickedObject) && Cesium.defined(pickedObject.label)) {
    pickedObject.label.show = false;
}
```

I realize that's more verbose, maybe ask around to see if that's what we really want.